src/ynEIGEN/TokenStakingNode.sol

```diff
@@ -150,20 +152,25 @@ contract TokenStakingNode is
 
         _fullWithdrawalRoots = tokenStakingNodesManager.delegationManager().queueWithdrawals(_params);
 
-        IYieldNestStrategyManager(tokenStakingNodesManager.yieldNestStrategyManager()).updateTokenStakingNodesBalances(
-            IERC20(address(0)),
-            _strategy
-        );
-
         emit QueuedWithdrawals(_strategy, _shares, _fullWithdrawalRoots);
     }
 
+    /**
+     * @notice Completes queued withdrawals for a specific strategy
+     * @param _nonce The nonce of the withdrawal
+     * @param _startBlock The block number when the withdrawal was queued
+     * @param _shares The number of shares to withdraw
+     * @param _strategy The strategy from which to withdraw
+     * @param _middlewareTimesIndexes The indexes of middleware times to use for the withdrawal
+     * @param updateTokenStakingNodesBalances If true calls updateTokenStakingNodesBalances for yieldNestStrategyManager
+     */
     function completeQueuedWithdrawals(
         uint256 _nonce,
         uint32 _startBlock,
         uint256 _shares,
         IStrategy _strategy,
-        uint256[] memory _middlewareTimesIndexes
+        uint256[] memory _middlewareTimesIndexes,
+        bool updateTokenStakingNodesBalances
```

Comment:
Since this should only have an effect in case of slashing, i suggest to make it optional so we can save some gas.

Also if we bundle completeQueuedWithdrawals with processPrincipalWithdrawals into one tx, like we did with ynETH (recent approach)
The computation would repeat anyway.